Core/include/Acts/Utilities/BinningType.hpp

```diff
@@ -41,19 +42,33 @@ enum BinningValue : int {
   binRPhi = 5,
   binH = 6,
   binEta = 7,
-  binMag = 8,
-  binValues = 9
+  binMag = 8
 };
 
-/// @brief static list of all binning values
-static const std::vector<BinningValue> s_binningValues = {
-    binX, binY, binZ, binR, binPhi, binRPhi, binH, binEta, binMag};
+/// Get all possible binning values
+/// @return the binning values
+const std::vector<BinningValue>& allBinningValues();
 
-/// @brief screen output option
-inline const std::vector<std::string>& binningValueNames() {
-  static const std::vector<std::string> _binningValueNames = {
-      "binX",    "binY", "binZ",   "binR",  "binPhi",
-      "binRPhi", "binH", "binEta", "binMag"};
-  return _binningValueNames;
+/// Returns the total number of binningvalues
+/// @return the number of binning values
+constexpr std::size_t numBinningValues() {
+  return 9;
```

Comment:
Is it essential that this function is `constexpr`? I wonder if this could be a potential pitfall in the future if we forget to update this value. If it does not need to be `constexpr` we could query the vector from the source files.